internal/services/network/virtual_network_resource.go

```diff
@@ -751,6 +954,45 @@ func resourceAzureSubnetHash(v interface{}) int {
 		if v, ok := m["security_group"]; ok {
 			buf.WriteString(v.(string))
 		}
+
+		if features.FourPointOhBeta() {
```

Comment:
Since we're changing this anyway, we should probably also take the opportunity to lower-case any Resource ID values when computing the hash value, to avoid any spurious changes?